libraries/chain/block_state.cpp

```diff
@@ -337,11 +337,28 @@ finality_data_t block_state::get_finality_data() {
    if (!base_digest) {
       base_digest = compute_base_digest(); // cache it
    }
+
+   // Check if there is any proposed finalizer policy in the block
+   std::optional<finalizer_policy> proposed_finalizer_policy;
+   if (is_savanna_genesis_block()) {
```

Comment:
If we are in a `block_state`, can this be the genesis block?

Reply:
A transition (and genesis) block can be reclassified as Savanna block. `get_transition_savanna_block()` does that.